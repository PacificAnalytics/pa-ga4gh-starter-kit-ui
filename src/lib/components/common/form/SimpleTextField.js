import React from 'react';
import { 
    FormControl,
    TextField
} from '@material-ui/core';

const SimpleTextField = props => {
    return(
        <FormControl fullWidth>
            <TextField
                id={props.id}
                label={props.label}
                margin='normal'
                name={props.name}
                type='text' 
                value={props.value}
                InputProps={{readOnly: props.readOnly}}
                onChange={e => props.changeFunction(e.target.value)}
                helperText={props.helperText}/>
        </FormControl>
    );
}

export default SimpleTextField;