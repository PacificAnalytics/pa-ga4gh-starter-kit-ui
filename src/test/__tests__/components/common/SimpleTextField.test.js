import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import SimpleTextField from '../../../../lib/components/common/SimpleTextField';
import {mockUpdateScalar} from '../../../resources/MockFunctions';

let simpleTextFieldProps = null;
beforeEach(() => {
    simpleTextFieldProps={
        id: 'SnapshotTestSimpleTextField',
        label: 'Snapshot Test',
        name: 'Snapshot Test',
        helperText: 'This is a description for a snapshot test.', 
        changeFunction: mockUpdateScalar
    }    
});

test('SHOW <SimpleTextField /> handles populated field', () => {
    let {container} = render(
        <SimpleTextField readOnly={true} value='Snapshot test value.'
        {...simpleTextFieldProps} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText(simpleTextFieldProps.label)).toBeInTheDocument();
    let simpleTextField = screen.getByRole('textbox');
    expect(simpleTextField).toHaveValue('Snapshot test value.');
    //typing does not result in the value being updated
    userEvent.type(simpleTextField, 'testing text entry');
    expect(mockUpdateScalar.mock.calls.length).toBe(0);
    expect(simpleTextField).toHaveAttribute('readonly');
});

test('NEW and EDIT <SimpleTextField /> handles populated field', () => {
    let {container} = render(
        <SimpleTextField readOnly={false} value='Snapshot test value.'
        {...simpleTextFieldProps} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText(simpleTextFieldProps.label)).toBeInTheDocument();
    let simpleTextField = screen.getByRole('textbox');
    expect(simpleTextField).toHaveValue('Snapshot test value.');
    userEvent.type(simpleTextField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
    expect(simpleTextField).not.toHaveAttribute('readonly');
}); 

test('NEW and EDIT <SimpleTextField /> handles empty field', () => {
    let {container} = render(
        <SimpleTextField readOnly={false} value='' {...simpleTextFieldProps} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText(simpleTextFieldProps.label)).toBeInTheDocument();
    let simpleTextField = screen.getByRole('textbox');
    expect(simpleTextField).toHaveValue('');
    userEvent.type(simpleTextField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
    expect(simpleTextField).not.toHaveAttribute('readonly');
});