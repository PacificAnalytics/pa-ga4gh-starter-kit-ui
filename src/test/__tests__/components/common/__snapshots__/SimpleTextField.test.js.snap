// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <SimpleTextField /> handles empty field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="SnapshotTestSimpleTextField"
      id="SnapshotTestSimpleTextField-label"
    >
      Snapshot Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="SnapshotTestSimpleTextField-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="SnapshotTestSimpleTextField"
        name="Snapshot Test"
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root"
      id="SnapshotTestSimpleTextField-helper-text"
    >
      This is a description for a snapshot test.
    </p>
  </div>
</div>
`;

exports[`NEW and EDIT <SimpleTextField /> handles populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="SnapshotTestSimpleTextField"
      id="SnapshotTestSimpleTextField-label"
    >
      Snapshot Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="SnapshotTestSimpleTextField-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="SnapshotTestSimpleTextField"
        name="Snapshot Test"
        type="text"
        value="Snapshot test value."
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="SnapshotTestSimpleTextField-helper-text"
    >
      This is a description for a snapshot test.
    </p>
  </div>
</div>
`;

exports[`SHOW <SimpleTextField /> handles populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="SnapshotTestSimpleTextField"
      id="SnapshotTestSimpleTextField-label"
    >
      Snapshot Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="SnapshotTestSimpleTextField-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="SnapshotTestSimpleTextField"
        name="Snapshot Test"
        readonly=""
        type="text"
        value="Snapshot test value."
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="SnapshotTestSimpleTextField-helper-text"
    >
      This is a description for a snapshot test.
    </p>
  </div>
</div>
`;
